### PR TITLE
fix: preserve todos after stream completes

### LINKED ISSUE #3271: To-dos 待办事项在流式传输期间可见，但在运行完成后会消失。在下一次对话流式传输期间又可见。
测试了最新代码，to-dos 列表在流式传输期间可以看到，在运行完成后消失。
再次发起对话，对话流式输出过程中，刚才的to-dos列表又可以看到。输出完成后又消失。

此为run1 对话结束后效果
Image

此为run2对话过程中效果

Image

### PR DESCRIPTION
**Pull request by 18062706139fcz**
## Summary
- preserve todos when a later thread state update omits the todos field
- keep explicit empty todo lists as a valid clear operation
- add regression tests for omitted-vs-empty todo state updates

## Verification
- pnpm exec vitest run tests/unit/core/threads/message-merge.test.ts
- pnpm exec eslint src/core/threads/hooks.ts tests/unit/core/threads/message-merge.test.ts
- pnpm typecheck

Closes #3271

### CODE REVIEW

**Comment by WillemJiang** [comment]
@18062706139fcz  could you take some time to resolve the conflict of the main branch?

File: frontend/src/core/threads/hooks.ts [hunk 1]
```diff
@@ -2374,10 +2396,17 @@ export function useThreadStream({
 
   // Merge history, live stream, and optimistic messages for display
   // History messages may overlap with thread.messages; thread.messages take precedence
+  const activeThreadId = onStreamThreadId ?? threadId ?? null;
+  if (valuesThreadIdRef.current !== activeThreadId) {
+    valuesThreadIdRef.current = activeThreadId;
+    valuesRef.current = thread.values;
+  }
+  const mergedValues = mergeThreadValues(valuesRef.current, thread.values);
```
Comment (willem-bd) [inline frontend/src/core/threads/hooks.ts:2491]: `mergedValues` gets a fresh object identity on every render: `mergeThreadValues` always returns a new spread, and `valuesRef.current = mergedValues` (line 2405) runs unconditionally. The SDK's `thread.values` is otherwise referentially stable across idle renders (`StreamManager.values` in `@langchain/langgraph-sdk` returns `this.state.values?.[0]`, which only changes when a stream update arrives), so this regresses identity stability for `mergedThread.values`.

Downstream consumers that list `thread.values` as a dependency now re-run on every render:
- `frontend/src/components/workspace/export-trigger.tsx:46` — `useCallback(..., [messages, thread.values, threadId, t])`
- `frontend/src/components/workspace/thread-title.tsx:39` — `useEffect(..., [..., thread.values])`

This is the same class of issue this file already guards against at lines 1737-1739 ("The SDK getter mints a fresh [] on every read ... normalize to a stable identity so downstream effects keyed on this array cannot re-fire (and setState-loop) on idle renders"). Consider only recomputing the merge when `thread.values` actually changes identity (e.g. track the last input in a ref and reuse the prior merged result on idle renders), so the todos fix doesn't trade for re-render storms in those consumers.

**Review by willem-bd — COMMENTED** [review]
Reviewed the todos-preservation fix. One referential-stability concern below; otherwise the merge logic looks sound and the omitted-vs-empty distinction is well-tested.
